web/views.py

```diff
@@ -200,18 +207,100 @@ def form_valid(self, form):
     """
 
+    form_class = forms.StudyListSearchForm
     template_name = "web/studies-list.html"
     model = Study
 
+    def post(self, request, *args, **kwargs):
+        return super().get(request, *args, **kwargs)
+
     def get_queryset(self):
-        # TODO if we need to filter by study demographics vs user demographics
-        # or by if they've taken the study before this is the spot
-        return super().get_queryset().filter(state="active", public=True).order_by("?")
+        user = self.request.user
+        qs = super().get_queryset().filter(state="active", public=True)
+        page = self.request.GET.get("page", 1)
+
+        # Form will return different value depending if the user is logged in.
+        form = self.search_form()
+
+        # values from form
+        search_value = form.data.get("search", "")
+        children_value = form.data.get("children", "")
+        show_experiments_already_done_value = (
+            form.data.get("show_experiments_already_done") == "on"
+        )
+
+        if search_value:
+            qs = qs.filter(name__icontains=search_value)
+
+        # convert to list as it's no longer being treated as a queryset
+        studies = list(qs)
+
+        if children_value:
+            if user.is_anonymous:
+                # when user is anonymous, children value is an age range.
+                age_range = [int(c) for c in children_value.split(",")]
+                studies = [
+                    s for s in studies if age_range_eligibility_for_study(age_range, s)
+                ]
+            else:
+                # when user is authenticated, children value is a child pk
+                child = Child.objects.get(pk=children_value)
+
+                if not show_experiments_already_done_value:
+                    studies = self.completed_consent_frame(studies, child)
+
+            studies = [s for s in studies if get_child_eligibility_for_study(child, s)]
+
+        self.child_eligibility(studies)
+
+        studies.sort(key=self.sort_fn())
+
+        return self.paginated_queryset(studies, page)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["search_form"] = self.search_form()
+        return context
+
+    def completed_consent_frame(self, studies, child):
+        return [
+            r.study
+            for r in Response.objects.filter(
+                study__in=studies, child=child, completed_consent_frame=True,
+            ).distinct("study_id")
+        ]
+
+    def child_eligibility(self, studies):
+        user = self.request.user
+
+        if user.is_authenticated:
+            children = user.children.filter(deleted=False)
+
+            # add eligible children to study object
+            for study in studies:
+                study.eligible_children = create_string_listing_children(
+                    [c for c in children if get_child_eligibility_for_study(c, study)]
+                )
```

Comment:
that's cool - how does it work to add stuff to the study that isn't part of what it'd have per the database? we can just add arbitrary fields?

Reply:
Python supports monkey patching.  As long as we don't override an existing value (that we might need) this should work well for us.